site/_posts/2021-09-16-how-to-authenticate-your-react-app.md

```diff
@@ -0,0 +1,1087 @@
+
+We’ll add our new login route to handle this:
+
+```bash
+touch server/routes/login.js
+```
+
+```
+server
+├─routes
+│ ├─login.js*
+│ └─user.js
+└─index.js
+```
+
+
+Whenever we add a new route, we need to let `server/index.js` know, just like we did with `/user`. In `server/index.js`:
+
+```js
+// ...
+
+// use routes
+var userRoute = (require('./routes/user'));
+var loginRoute = (require('./routes/login'));
+
+// ...
+```
+
+Your entire `server/index.js` file should look like this:
+
+```js
+var userRoute = (require('./routes/user'));
+var loginRoute = (require('./routes/login'));
+const express = require('express');
+const cors = require('cors');
+const config = require('../client/src/config');
+const PORT = process.env.PORT || 3001;
+
+
+// configure Express app and install the JSON middleware for parsing JSON bodies
+const app = express();
+app.use(express.json());
+
+app.use(cors({
+  origin: true,
+  credentials: true
+}));
+
+// use routes
+app.use('/user', userRoute);
+app.use('/login', loginRoute);
+
+// start server
+app.listen(config.serverPort, () => console.log(`FusionAuth example app listening on port ${config.serverPort}.`));
+
+```
+
+When a user goes to `/login`, we redirect them from the Express sever to FusionAuth’s login page. One of the main reasons we installed FusionAuth is so that we don’t have to handle the login process at all!  
+
+*Note: there is a way to access the FusionAuth pages in an iframe so that the user doesn't have to be redirected, but that is outside the scope of this tutorial.*
+
+Now, lets configure our `server/routes/login.js` file to do just that:
+
+```js
+const express = require('express');
+const router = express.Router();
+const config = require('../../client/src/config');
+
+router.get('/', (req, res) => {
+  res.redirect(`http://localhost:${config.fusionAuthPort}/oauth2/authorize?client_id=${config.clientID}&redirect_uri=${config.redirectURI}&response_type=code`);
+});
+
+module.exports = router;
+```
+
+If that URI looks a bit messy, it’s because of the additional query parameters, which FusionAuth needs to process our request:
+
+- `client_id` tells FusionAuth which app is making the request
+- `redirect_uri` tells FusionAuth where to redirect the user to after login
+- `response_type` tells FusionAuth which OAuth grant we’re using (Authorization Code in this example)
+
+This is all part of the standard OAuth Authorization Code grant.
+
+Try navigating to `localhost:3001/login`. If you see a FusionAuth login form, you're all set!
+
+{% include _image.liquid src="/assets/img/blogs/fusionauth-example-react-2021/login.png" alt="login screen" class="img-fluid" figure=false %}
```

Comment:
Should crop this to omit all the whitespace at the bottom. Can show you how to do this in preview (draw a box, hit command-k).